fontforgeexe/uiutil.c

```diff
@@ -528,27 +535,31 @@ return( GGadgetDispatchEvent(errdata.vsb,event));
       case et_expose:
 	  /*GDrawFillRect(gw,&event->u.expose.rect,GDrawGetDefaultBackground(NULL));*/
 	  GDrawSetFont(gw,errdata.font);
-	  for ( i=0; i<errdata.linecnt && i+errdata.offtop<errdata.cnt; ++i ) {
+	  s_l = errdata.start_l, s_c = errdata.start_c, e_l = errdata.end_l, e_c = errdata.end_c;
+	  if ( s_l>e_l ) {
+		  s_l = e_l; s_c = e_c; e_l = errdata.start_l; e_c = errdata.start_c;
+	  }
+	  if (s_l == e_l && s_c > e_c) {
+		  s_c = e_c;
+		  e_c = errdata.start_c;
+	  }
+	  for ( i=0, j = errdata.offtop; i<errdata.linecnt && j<errdata.cnt; ++i, ++j ) {
 	      int xs, xe;
-	      int s_l = errdata.start_l, s_c = errdata.start_c, e_l = errdata.end_l, e_c = errdata.end_c;
 	      GRect r;
-	      if ( s_l>e_l ) {
-		  s_l = e_l; s_c = e_c; e_l = errdata.start_l; e_c = errdata.start_c;
-	      }
-	      GDrawLayoutInit(gw,errdata.errlines[i+errdata.offtop],-1,NULL);
-	      if ( i+errdata.offtop >= s_l && i+errdata.offtop <= e_l ) {
-		  if ( i+errdata.offtop > s_l )
+	      GDrawLayoutInit(gw,errdata.errlines[j],-1,NULL);
+	      if ( j >= s_l && j <= e_l && (j < e_l || e_c > 0) ) {
+		  if ( j > s_l )
 		      xs = 0;
 		  else {
 		      GRect pos;
 		      GDrawLayoutIndexToPos(gw,s_c,&pos);
 		      xs = pos.x+3;
 		  }
-		  if ( i+errdata.offtop < e_l )
+		  if ( j < e_l )
 		      xe = 3000;
 		  else {
 		      GRect pos;
-		      GDrawLayoutIndexToPos(gw,s_c,&pos);
+		      GDrawLayoutIndexToPos(gw,e_c-1,&pos);
```

Comment:
Not sure I understand this change, but not in the sense that I think something is wrong with it.

Reply:
Basically this highlight is inclusive, but the copy-to-clipboard routine uses the end column as exclusive, so it's just aligning the two so that what you select matches what gets copied.

Reply:
So was it just broken before?

Reply:
Well originally there was a typo so it was using `s_c` instead, so you'd see the end position tracking your cursor when selecting text. So yeah it was pretty broken

Reply:
Ah, good -- the switch was what I didn't understand. 

I've already approved but consider this PR fully approved at this point.

Reply:
Oh , I guess I should have read the description better. *sigh*